File: po-rest/src/main/java/org/tradefinance/main/services/ShipmentService.java
package org.tradefinance.porest.services;

import java.util.Collection;

import org.tradefinance.assets.Shipment;

public interface ShipmentService extends org.tradefinance.common.services.ShipmentService {

    public Collection<Shipment> getShipments(String behalfOfId) throws Exception;

    public Shipment getShipment(String id) throws Exception;

    public Shipment getShipmentByHash(String id) throws Exception;

    public Shipment createShipment(String purchaseOrderId, int units, String senderId, String receiverId) throws Exception;

    public void deliverShipment(String id) throws Exception;
}

File: cli/po-cli/src/services/shipmentservice.ts
import { HttpService } from 'wt-common-cli';

export class ShipmentService {
    httpService: HttpService;

    constructor(httpService: HttpService) {
        this.httpService = httpService;
    }

    async createShipment(purchaseOrderId: string, units: number, senderId: string, receiverId: string) {
        const shipmentBody = {
            purchaseOrderId,
            units,
            senderId,
            receiverId
        };

        const partialShipment = await this.httpService.post('shipments', shipmentBody);

        return this.getShipment(partialShipment.id);
    }

    async getShipment(id: string) {
        return this.httpService.get(`shipments/${id}`);
    }

    async getShipments(user: string) {
        return this.httpService.get(`shipments?behalfOf=${user}`);
    }
}
